src/DiffOp/DiffOpAdd.php

```diff
@@ -64,8 +76,20 @@ public function serialize() {
 	 *
 	 * @param string $serialization
 	 */
+	#[\ReturnTypeWillChange]
 	public function unserialize( $serialization ) {
-		$this->newValue = unserialize( $serialization );
+		$this->__unserialize( unserialize ($serialization) );
```

Comment:
Isn’t this an incompatible change to the serialization? Previously, `unserialize( $serialization )` was directly the `newValue`, which could apparently be anything; now, it’s expected to be an array *containing* the `newValue`. If the new code tries to deserialize a value serialized by the old code, what happens?

Reply:
Isn't the same case here? https://github.com/wikimedia/mediawiki/commit/4cc1a3eecaaecc4368753a9a08fa0732bddae6fb

Reply:
Do you have a specific class in mind? At a glance, I don’t see the same problem in that commit.